Assets/Scripts/Critter.cs

```diff
@@ -0,0 +1,108 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+/// <summary>
+/// This is the base class of all critters.
+/// </summary>
+public abstract class Critter
+{
+    // DISPLAY
+    [SerializeField]
+    GameObject _display;
+    
+    // STATS
+    public string Name { get; }
+    public int BaseHealth { get; }
+    public int BaseEnergy { get; }
+    public int BaseSpeed { get; }
+
+    /// <summary>
+    /// This is the list of actions your critter can perform in combat.
+    /// </summary>
+    public virtual List<UnityAction> Moveset { get; private set; }
+
+    // CURRENT VALUES
+    public int Health { get; private set; }
+    public int Energy { get; private set; }
+    public int Speed { get; set; }
+
+    public List<StatusEffect> statusEffects = new ();
+
+    
+    // EVENTS
+    public abstract void Act(ICritterBattleManager bm);
+
+    public virtual void OnTurn()
+    {
+        statusEffects.ForEach(e => e.AffectOnTurn());
+        // Remove all effects whose timer has reached 0.
+        statusEffects.RemoveAll(StatusEffect.RemoveCheck);
+        
+        // Display / update effect list.
+    }
+
+    
+    public virtual void OnHurt(Attack receiving, bool applyEffects = true)
```

Comment:
Do we need the optional bool param here? (I'd keep things simple unless they're needed, especially for optional params)